src/main/java/tweet/InsertDB2.java

```diff
@@ -0,0 +1,392 @@
+               lang = x[1];
+
+               if (x[0].equals("poi")) {
+                  continue;
+               }
+
+               for (int i = 0; i < len; i++) {
+                  temp[i] = temp[i].trim();
+               }
+               if (len == 4) {
+                  if (x[0].equals("city") || x[0].equals("admin")) {
+                     if (temp[1].equals(temp[0])) {
+                        continue;
+                     }
+                     city = temp[0];
+                     state = null;
+                     country = temp[1];
+                  } else if (x[0].equals("country") || x[0].equals("neighborhood")) {
+                     continue;
+                  }
+               } else if (len == 5) {
+                   if (x[0].equals("city")) {
+                     if ((temp[1]).equals(temp[2])) {
+                        city = temp[0];
+                        state = null;
+                        country = temp[2];
+                     } else {
+                        city = temp[0];
+                        state = temp[1];
+                        country = temp[2];
+                     }
+                  } else if (x[0].equals("coords")) {
+                     city = temp[0];
+                     state = temp[1];
+                     country = temp[2];
+                  } else if (x[0].equals("neighborhood")) {
+                     try {
+
+                        final ArrayList<String> states = countryState.get(temp[2]);
+                        if (states.contains(temp[1])) {
+                           city = temp[0];
+                           state = temp[1];
+                           country = temp[2];
+                        } else {
+                           city = temp[1];
+                           state = null;
+                           country = temp[2];
+                        }
+                     } catch (final NullPointerException e) {
+                        /* Case when country doesn't exist in the country-state
+                        dictionary (because of spelling/language difference maybe)
+                        and .contains function throws a Null pointer exception
+                        */
+                        city = temp[1];
+                        state = null;
+                        country = temp[2];
+                     }
+                  } else if (x[0].equals("admin")) {
+
+                     if ((temp[2]).equals(temp[1])) {
+                        try {
+                           final ArrayList<String> states = countryState.get(temp[2]);
+                           if (states.contains(temp[0])) {
+                              continue;
+                           } else {
+                              city = temp[0];
+                              state = null;
+                              country = temp[2];
+                           }
+                        } catch (final NullPointerException e) {
```

Comment:
You should check for `null` instead. Throwing and catching exceptions should not replace normal control flow.